src/Compilers/Test/Core/Compilation/RuntimeUtilities.cs

```diff
@@ -27,6 +27,9 @@ public static partial class RuntimeUtilities
 #endif
         internal static bool IsCoreClrRuntime => !IsDesktopRuntime;
 
+        internal static bool IsCoreClr6Runtime
+            => IsCoreClrRuntime && RuntimeInformation.FrameworkDescription.StartsWith(".NET 6.");
```

Comment:
is there a check we can do practically which will be "durable" for .NET 7 and beyond?

Reply:
I'm open to suggestions on how to better slice this.
Our configuration for running `dotnet test` ensures that we're on .NET 6 or later. So the CoreClr case that's not ".NET 6" effectively means ".NET 7 or later".

Reply:
Make sure to use an explicit comparer here (string comparison changes with runtime)

Reply:
Could we parse the description confidently enough to get the Runtime Major version? That might be useful down the road, as well as making the version checks more meaningful.

Reply:
@JoeRobich I'm hesitant to engage in parsing because the format is not that well [specified](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription).